File: frontend/src/types/report.ts
/**
 * Violation-report types — mirror the /api/reports endpoints.
 *
 * A report is a deliberate escalation of a flagged scan with its own review
 * lifecycle, so it is distinct from a scan.
 */

export type ReportStatus = "submitted" | "under_review" | "resolved" | "dismissed";

export const REPORT_STATUSES: ReportStatus[] = [
  "submitted",
  "under_review",
  "resolved",
  "dismissed",
];

export const REPORT_STATUS_LABELS: Record<ReportStatus, string> = {
  submitted: "Submitted",
  under_review: "Under Review",
  resolved: "Resolved",
  dismissed: "Dismissed",
};

export interface ReportViolation {
  rule_id: number | null;
  field_name: string | null;
  clause_reference: string | null;
  description: string | null;
  extracted_value: string | null;
}

export interface ReportItem {
  report_id: string;
  scan_id: string;
  reported_by: string | null;
  status: ReportStatus;
  notes: string | null;
  email_sent: boolean;
  created_at: string;
  updated_at: string;
  product_name: string | null;
  brand: string | null;
  category: string | null;
  violations: ReportViolation[];
}

export interface ReportListResponse {
  items: ReportItem[];
  total: number;
}

export interface ReportFileResponse {
  report_id: string;
  scan_id: string;
  status: string;
  violations: ReportViolation[];
  email_sent: boolean;
  created_at: string;
  updated_at: string;
  product_name: string | null;
  brand: string | null;
  category: string | null;
}

export type ReportListItem = ReportItem;
export type ReportDetail = ReportItem;

File: backend/app/schemas/report.py
"""Pydantic schemas for Violation Reports."""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class ReportViolationItem(BaseModel):
    rule_id: Optional[int] = None
    field_name: Optional[str] = None
    clause_reference: Optional[str] = None
    description: Optional[str] = None
    extracted_value: Optional[str] = None


class ReportCreateRequest(BaseModel):
    scan_id: str
    reported_by: Optional[str] = None
    notes: Optional[str] = None


class ReportStatusUpdate(BaseModel):
    status: str  # submitted, under_review, resolved, dismissed


class ReportResponse(BaseModel):
    report_id: str
    scan_id: str
    reported_by: Optional[str] = None
    status: str
    notes: Optional[str] = None
    email_sent: bool = False
    created_at: datetime
    updated_at: datetime
    product_name: Optional[str] = None
    brand: Optional[str] = None
    category: Optional[str] = None
    violations: List[ReportViolationItem] = []

    model_config = ConfigDict(from_attributes=True)


class ReportListResponse(BaseModel):
    items: List[ReportResponse]
    total: int
